DataMigration/PostgresDB/IDataProvider.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+using DataMigration.PostgresDB.Entities;
+
+namespace DataMigration.PostgresDB
+{
+    interface IDataProvider
+    {
+        List<HistoricalOcrData> GetHistoricalOcrData(int? count = null, bool deleteAfterSelect = false);
```

Comment:
it is not proper architectural decision to add a delete (or any modifying) logic into get method